### ISSUE TITLE
Error when running for the first time after installing.

### ISSUE THREAD

**Issue by Gronis** [glymehrvrd/CppYCM#10]
Cannot install this plugin properly. When launching Sublime Text, I get this error prompt, and then python crashes. I'm using python 2.7.6
```
reloading Packages/User/C++YouCompleteMe.sublime-settings
Traceback (most recent call last):
  File "./urllib/request.py", line 1248, in do_open
  File "./http/client.py", line 1065, in request
  File "./http/client.py", line 1103, in _send_request
  File "./http/client.py", line 1061, in endheaders
  File "./http/client.py", line 906, in _send_output
  File "./http/client.py", line 844, in send
  File "./http/client.py", line 822, in connect
  File "./socket.py", line 435, in create_connection
  File "./socket.py", line 426, in create_connection
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 312, in on_activated_async
    callback.on_activated_async(v)
  File "listeners.loadextraconf in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 40, in on_activated_async
  File "lib.ycmd_handler in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 31, in server
  File "lib.ycmd_handler in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 181, in WaitUntilReady
  File "lib.ycmd_handler in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 99, in IsReady
  File "lib.ycmd_handler in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 114, in GetFromHandler
  File "lib.ycmd_handler in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 200, in _CallHttp
  File "./urllib/request.py", line 156, in urlopen
  File "./urllib/request.py", line 469, in open
  File "./urllib/request.py", line 487, in _open
  File "./urllib/request.py", line 447, in _call_chain
  File "./urllib/request.py", line 1268, in http_open
  File "./urllib/request.py", line 1251, in do_open
urllib.error.URLError: <urlopen error [Errno 61] Connection refused>
Exception in thread LoadExtraConfAsync:
Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "./threading.py", line 858, in run
  File "listeners.loadextraconf in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 22, in load_extra_conf_func
  File "lib.ycmd_handler in /Users/robin/Library/Application Support/Sublime Text 3/Installed Packages/C++YouCompleteMe.sublime-package", line 179, in WaitUntilReady
RuntimeError: waited for the server for 5 seconds, aborting
```

**Comment by glymehrvrd** [comment]
The lastest YCMD server changed HMAC calculation(Valloric/ycmd/#113) while this plugin doesn't update to match it, so you got the `Connection refused` error. But if I update this plugin to match the new HMAC calculation, the ones who use the older YCMD server will get into trouble. I'll try to solve this problem when I'm not so busy.
You can use the older version of YCMD or fork this plugin and update the HMAC calculation yourself(See 426833360adec8db72ed6cef9d7aa7f037e6a5b8)
:)

**Comment by Gronis** [comment]
Hi :) Thanks for the reply, though I'm still not getting this to work. I've tried with compiling the development branch of ycmd (Updated dec 2014) without the changes you mentioned. This still gives me the same error output.

**Comment by glymehrvrd** [comment]
It should be solved, see commit 7adfc1012234992d6ec2794768f69deac4b273cc.

**Comment by Gronis** [comment]
I can confirm that this fix works. (at least on my OSX computer)
